snapshotter/benchmark/benchmark_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package benchmark
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/containerd/containerd/mount"
+	"github.com/containerd/containerd/snapshots"
+	"github.com/containerd/containerd/snapshots/overlay"
+	"github.com/containerd/continuity/fs/fstest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/firecracker-microvm/firecracker-containerd/snapshotter/devmapper"
+	"github.com/firecracker-microvm/firecracker-containerd/snapshotter/naive"
+)
+
+var (
+	dmPoolDev       string
+	dmRootPath      string
+	overlayRootPath string
+	naiveRootPath   string
+)
+
+func init() {
+	flag.StringVar(&dmPoolDev, "dm.thinPoolDev", "", "Pool device to run benchmark on")
+	flag.StringVar(&dmRootPath, "dm.rootPath", "", "Root dir for devmapper snapshotter")
+	flag.StringVar(&overlayRootPath, "overlay.rootPath", "", "Root dir for overlay snapshotter")
+	flag.StringVar(&naiveRootPath, "naive.rootPath", "", "Root dir for naive snapshotter")
```

Comment:
The idea here is to run each snapshotter on separate EBS volume, so they have exactly same conditions.

Reply:
Also I keep three path variables instead of just one for convenience, so you can run all benchmarks with one command and get unified output in just one table. Theoretically we can have one variable for this, but you have to run each benchmark separately.